libraries/tools/kotlin-gradle-plugin-integration-tests/src/test/kotlin/org/jetbrains/kotlin/gradle/testbase/testDsl.kt

```diff
@@ -231,30 +231,33 @@ internal fun Path.enableCacheRedirector() {
             """.trimIndent()
         )
 
-    when {
-        resolve("build.gradle").exists() -> {
-            //language=Groovy
-            resolve("build.gradle").appendText(
-                """
-                
-                allprojects {
-                    apply from: "${'$'}rootDir/gradle/cacheRedirector.gradle.kts"
-                }
-                
-                """.trimIndent()
-            )
-        }
-        resolve("build.gradle.kts").exists() -> {
-            //language=Groovy
-            resolve("build.gradle.kts").appendText(
-                """
-                
-                allprojects {
-                    apply(from = "${'$'}rootDir/gradle/cacheRedirector.gradle.kts")
-                }
-                
-                """.trimIndent()
-            )
+    val projectDir = toFile()
+    projectDir.walk().forEach {
```

Comment:
Why such change is required? `allprojects {}` is not compatible with project isolation? :thinking:

Reply:
Yup, it is not compatible cause you are accessing subproject from root project. See https://gradle.github.io/configuration-cache/